### PR TITLE
#4179 - Clicking check connection the sync URL is changed

### LINKED ISSUE #4170: "Check connection" causes sync to halt and further shows inconsistent behavior
## Describe the bug

on clicking Settings -> Check connection, causes sync to fails. 
### To reproduce

1. Go to Settings
2. Click on Check Connection. It shows message "Connection successful" if working properly
3. Do some requisition and manually sync the mobile site
4. See error. Sync fails with the message "Sync record rejected....."
5.  Now logout from the system.
6. login again
7. See Error. "Site name not found" error shown and one couldn't login into the site

https://user-images.githubusercontent.com/58979436/123241244-b054dd00-d500-11eb-8508-85d47d265919.mp4

### Expected behaviour

Sync should be consistent in any operation

### Proposed Solution

Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info

- App version: V8.1.0
- Tablet model: Samsung
- OS version: 8.1

### LINKED ISSUE #4179: Send requisition lines even when requested quantity = 0 (if AMC > 0 or Available stock > 0)
## Is your feature request related to a problem? Please describe.

IC District stores want to be able to have visibility of this data to trigger a replenishment if needed.

## Describe the solution you'd like

As per description

## Implementation

- Add a store pref to toggle this customisation on/off
- (If above pref is enabled) When a requisition is finalised or an automatic order is generated, only prune lines if AMC and Available Stock are both 0, otherwise include them.
- (If above pref is enabled) Default 'Hide over stocked' toggle to `false`, i.e. show the overstocked lines.

## Describe alternatives you've considered

Not implementing?

## Additional context

**Add any other context or screenshots about the feature request here.**

### PR DESCRIPTION
**Pull request by bijaySussol**
Fixes #4170

## Change summary

Explain and justify your changes

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Setup server with mobile version 8.1.0
- [ ] Connect mobile with server
- [ ] Once you are on home page of Mobile
- [ ] Click on Setting
- [ ] Click on `Check connection`
   - [ ] It should alert depending on the status of Sync URL
   - [ ] go back to home page and click on setting
      - [ ] Sync URL should not be changed to something like `SyncURL/sync/v3/site`  

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

**Comment by bijaySussol** [comment]
@joshxg : 
Could you please help to review?
I did the changes as i understood and my understanding is this:

`
As far as i understood after checking - We do not need to add a end point '/sync/v3/site' - as we imported the method inside SettingPage - if we just pass the current URL - The end point `/sync/v3/site` will be handled by method syncAuthenticator
`

File: src/pages/SettingsPage.js [hunk 1]
```diff
@@ -81,11 +81,11 @@ const Settings = ({
 
   const onCheckConnection = () => {
     //  Hash the password.
-    const authURL = `${syncURL}${AUTH_ENDPOINT}`;
+
     const syncSiteName = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
     const syncAuthenticator = new SyncAuthenticator(AppSettings);
     syncAuthenticator
-      .authenticate(authURL, syncSiteName, null, currentUserPasswordHash)
+      .authenticate(syncURL, syncSiteName, null, currentUserPasswordHash)
```
Comment (joshxg) [inline src/pages/SettingsPage.js:88]: Sorry, we also need to change this password.. should be `UIDatabase.getSetting(SETTING_KEY.SYNC_SITE_PASSWORD_HASH)`

**Review by joshxg — CHANGES_REQUESTED** [review]

File: src/pages/SettingsPage.js (hunk 1 above)
Comment (bijaySussol) [reply src/pages/SettingsPage.js]: No worries.
This ae94f46 should make you happy.

**Comment by bijaySussol** [comment]
@joshxg : 
Requested changes are done.
Re-requesting for Review.

**Review by joshxg — APPROVED** [review]
Cool- I've changed the base to master so we can look at getting this into a patch soon. Will merge it soon!

**Comment by joshxg** [comment]
@katherine-sussol maybe into 8.1.1 also?

**Pull request merged**